podman/domain/image_build.py

```diff
@@ -155,3 +158,18 @@ def build(self, **kwargs) -> Tuple[Image, Iterator[bytes]]:
             return self.get(image_id), report_stream
 
         raise BuildError(unknown or 'Unknown', report_stream)
+
+    @staticmethod
+    def _build_context(dockerfile: Union[io.BytesIO, str]) -> IO:
+        file = tempfile.NamedTemporaryFile()
+        with tarfile.open(mode="w", fileobj=file) as tarball:
+            if isinstance(dockerfile, io.BytesIO):
+                info = tarfile.TarInfo("Dockerfile")
+                info.size = len(dockerfile.getvalue())
+                dockerfile.seek(0)
+            else:
+                info = tarball.gettarinfo(fileobj=dockerfile, arcname='Dockerfile')
+            tarball.add(info, dockerfile)
+
+        file.seek(0)
+        return file
```

Comment:
PITA time, can we mostly convert Dockerfile to Containerfile throughout?  Perhaps not in this PR, but at some point in the not too distant future?  Not only messaging but also variables within?  Or perhaps make it secondary to Containerfile?   (Tom quickly digs fox hole to hide from Jhon in :bomb: )

Reply:
I can change variables and default values, I cannot change the keyword arguments as that would break developers expectations.